config/checkstyle-examples-checks.xml

```diff
@@ -61,12 +50,50 @@
     <property name="minimum" value="1"/>
     <property name="maximum" value="1"/>
   </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="forbiddenRegexpInViolationMessage"/>
+    <property name="format" value="^.*//.* violation.*(\.\*|\.\+).*$"/>
+    <property name="message"
+              value="Forbidden usage of regex: .* and .+ are not allowed"/>
+    <property name="maximum" value="0"/>
+  </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="discouragedRegexpInViolationMessage"/>
+    <property name="format" value="^.*//.* violation.*\\[\.\+\*\?\|\{\}\(\)\[\]\^\-\$].*$"/>
+    <property name="message"
+              value="Discouraged usage of escaped special regex symbols. Please ensure violation
+                     message really needs this symbol"/>
+    <property name="maximum" value="0"/>
+  </module>
```

Comment:
Sad that we need to compromise clarity/readability in documentation when we do not even know if escaped regexp tokens will be a problem for us in documentation generation.